File: src/skid_steer_py_examples/skid_steer_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimpleSubscriber(Node):
    def __init__(self):
        super().__init__("Simple_subscriber")
        self.sub = self.create_subscription(String, "Chatter", self.MsgCallback, 10)
    def MsgCallback(self, msg):
        self.get_logger().info("I heard: %s" % msg.data)
   
def main():
    rclpy.init()
    simple_subscriber = SimpleSubscriber()
    rclpy.spin(simple_subscriber)
    simple_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__name__':
    main()

File: src/skid_steer_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
using std::placeholders::_1;
class SimpleSubscriber : public rclcpp::Node{
public:
    SimpleSubscriber() : Node("Simple_subscriber"){
       sub = create_subscription<std_msgs::msg::String>("Chatter", 10, std::bind(&SimpleSubscriber::MsgCallback, this, _1));
    }
private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub;
    void MsgCallback(const std_msgs::msg::String &msg) const{
        RCLCPP_INFO_STREAM(get_logger(), "I heard3232: " << msg.data.c_str());
    }
};

int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}